### DISCUSSION TITLE
Using the plugin behind a reverse proxy

### DISCUSSION THREAD

**Discussion by derekantrican** [coddingtonbear/obsidian-local-rest-api#66] [Show and tell]
Just want to share something I finally got working after #60 was resolved via #65.

I wanted to set up the Obsidian local rest API behind a reverse proxy (in my case: nginx). I have an `auth_basic` specification so an `Authorization ...` header is already required for interacting with nginx - meaning that the above linked changes were necessary as there isn't really precedent for multiple `Authorization` headers (there's some hacky workarounds, but nginx can't really support them). Thanks @coddingtonbear !

So using the new Advanced Settings option, I changed the `Authorization` header to `X-API-KEY` and after a bit of trial & error, set up the following location block:

```nginx
location /obsidian {
    rewrite ^/obsidian/?(.*)$ /$1 break;
    proxy_pass http://127.0.0.1:27123;
    proxy_set_header Host $host;
    proxy_set_header X-API-KEY "Bearer [TOKEN]";
    proxy_redirect off;
}
```
Obviously `[TOKEN]` is where I've put my token from the plugin. You could remove this line and require the call to your server to pass both the `Authorization ...` header for nginx AND the `X-API-KEY ...` header for the Obsidian plugin.

Since the plugin doesn't have a "base url" option to work with reverse proxies, I had to use the `rewrite` rule. This will turn something like `http://myserver.com/obsidian/search/simple?query=test` into `http://127.0.0.1:27123/search/simple?query=test` locally (removing the `/obsidian` section).

Hope this helps someone else!

**Comment by motion2082** [comment]
For anyone trying to get this working on Cloudflare Reverse Tunnel, turn off TLS Verify and create a Port Proxy on your Windows machine
